xrviz/fields.py

```diff
@@ -85,23 +84,46 @@ def setup(self, var):
             else:
                 self.y.options = y_opts
                 self.remaining_dims = [opt for opt in y_opts if opt!=self.y.value]
-        self.change_dim_selectors()
+                self.change_y()
 
-    def change_y(self, value):
+    def change_y(self, value=None):
         """
         Updates the options of y, by removing option selected in x (value),
         from all the variable dimensions available as options.
         """
-        values = self.var_dims.copy()
-        values.remove(self.x.value)
-        self.y.options = values
+         # if x belong to var_dims replace the y with remaining var_dims
+        # else if x belong to non_indexed_coords, replace y with remaining
+        # non_indexed_coords
+        values = self.sel_options.copy()
+        x_val = self.x.value
+        values.remove(x_val)
+        if isinstance(self.data, xr.Dataset):
+            if x_val in self.var_dims:
+                valid_values = list(set(values) - set(self.non_indexed_coords))
```

Comment:
Some more list->set->list with operators; perhaps we need some utils helper functions.

Reply:
I have simplified this.